### PR TITLE
Use target_size_t for VNForByrefCon

### LINKED ISSUE #13745: valuenum should use `target_size_t` for `TYP_BYREF`
There are ~7 calls to `VNForByrefCon(size_t cnsVal)` and they should be changed into `VNForByrefCon(target_size_t cnsVal)` calls.

That change will convert 64bit constants to 32bit constants for byrefs when doing cross-crossgen for arm32 on arm64 . We should see diffs on the test added in dotnet/coreclr#27702.

_Originally posted by @echesakovMSFT in https://github.com/dotnet/coreclr/pull/27702_

category:correctness
theme:ir
skill-level:beginner
cost:small

### PR DESCRIPTION
**Pull request by briansull**
Fixes #13745

### CODE REVIEW

**Comment by briansull** [comment]
@sandreenko PTAL
cc: @dotnet/jit-contrib

File: src/coreclr/jit/valuenum.cpp [hunk 1]
```diff
@@ -2675,7 +2675,7 @@ ValueNum ValueNumStore::EvalFuncForConstantArgs(var_types typ, VNFunc func, Valu
             {
                 assert((typ == TYP_BYREF) || (typ == TYP_LONG));
```
Comment (sandreenko) [inline src/coreclr/jit/valuenum.cpp:2676]: can the assert be changed to `assert((typ == TYP_BYREF) || (typ == TYP_I_IMPL));` so we are sure we don't cast LONG to 32-bit on x86/arm32?

File: src/coreclr/jit/valuenum.cpp [hunk 2]
```diff
@@ -2675,7 +2675,7 @@ ValueNum ValueNumStore::EvalFuncForConstantArgs(var_types typ, VNFunc func, Valu
...
                 size_t resultVal = EvalOp<size_t>(func, arg0Val, arg1Val);
-                result           = VNForByrefCon(resultVal);
+                result           = VNForByrefCon((target_size_t)resultVal);
```
Comment (sandreenko) [inline src/coreclr/jit/valuenum.cpp:2678]: Nit: Use `static_cast<>` here and below to conform with https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#1511-casts

**Review by sandreenko — COMMENTED** [review]
LGTM, thanks for fixing this.

**Review by sandreenko — APPROVED** [review]

File: src/coreclr/jit/valuenum.cpp (hunk 2 above)
Comment (echesakovMSFT) [reply src/coreclr/jit/valuenum.cpp:2678]: Shouldn't we update types of `arg0Val` and `arg1Val1` to be `target_size_t`?

File: src/coreclr/jit/valuenum.cpp (hunk 1 above)
Comment (briansull) [reply src/coreclr/jit/valuenum.cpp:2676]: OK

File: src/coreclr/jit/valuenum.cpp (hunk 2 above)
Comment (briansull) [reply src/coreclr/jit/valuenum.cpp:2678]: Currently we aren't using static_cast<> in the valuenum.cpp.
If this is desirable then we can open a cleanup issue to change to use static_cast's

Comment (briansull) [reply src/coreclr/jit/valuenum.cpp:2678]: I would rather not change the types of arg0Val and argt1Val here

**Pull request merged**